DigitalLearningSolutions.Data.Tests/Services/TutorialContentServiceTests.cs

```diff
@@ -582,6 +668,402 @@ public void Get_tutorial_information_should_have_default_course_settings_when_js
+        }
+
+        [Test]
+        public void Get_tutorial_information_should_not_have_otherSectionsExist_when_other_section_has_no_diagnostic_assessment_path()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 5694;
+                const int candidateId = 1;
+                const int sectionId = 103;
+                const int tutorialId = 322;
+
+                // Make this tutorial viewable
+                tutorialContentTestHelper.UpdateTutorialStatus(tutorialId, customisationId, 1);
+
+                // Remove diagnostic assessment paths from other sections
+                int[] otherSections = { 104, 105, 106, 107, 108, 109, 110, 111 };
+                otherSections.ToList().ForEach(section =>
+                    tutorialContentTestHelper.UpdateDiagnosticAssessmentPath(section, null)
+                );
+
+                // When
+                var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+                // Then
+                tutorial.Should().NotBeNull();
+                tutorial!.OtherSectionsExist.Should().BeFalse();
+            }
+        }
+
+        [Test]
+        public void Get_tutorial_information_should_have_otherSectionsExist_when_other_sections_only_have_post_learning_assessment()
+        {
+            // Given
+            const int customisationId = 10820;
+            const int candidateId = 1;
+            const int sectionId = 104;
+            const int tutorialId = 326;
+
+            // When
+            var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+            // Then
+            tutorial.Should().NotBeNull();
+            tutorial!.OtherSectionsExist.Should().BeTrue();
+        }
+
+        [Test]
+        public void Get_tutorial_information_should_not_have_otherSectionsExist_when_other_section_has_no_post_learning_assessment_path()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int customisationId = 10820;
+                const int candidateId = 1;
+                const int sectionId = 104;
+                const int tutorialId = 326;
+
+                // Remove post learning assessment paths from other sections
+                int[] otherSections = { 103, 105, 106, 107, 108, 109, 110, 111 };
+                otherSections.ToList().ForEach(section =>
+                    tutorialContentTestHelper.UpdatePostLearningAssessmentPath(section, null)
+                );
+
+                // When
+                var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+                // Then
+                tutorial.Should().NotBeNull();
+                tutorial!.OtherSectionsExist.Should().BeFalse();
+            }
+        }
+

```

Comment:
This file is getting pretty long! I think it's worth splitting it up. You could use partial classes to still share the setup (e.g. like the learning portal controller tests). E.g. one file for the main tests, one for the next button fields and one for the other tutorial/section fields. What do you think? 🙂

Reply:
Yep I agree. I've split them up in d8f75b along those lines, though I wonder if the other tutorial/section and next tutorial/section test files can be each split again. What do you think? 🙂

Reply:
Up to you. I think the files are a reasonable size now so fine to leave as is. But also happy if you want to split them up further 🙂 Worth grouping them all together in a folder though (if you haven't done that already) 🙂

Reply:
Yep they are in a folder. I think I'll leave them as they are then 🙂